backend/internal/observability/subscriber/defaultsubscriber/default_subscriber.go

```diff
@@ -0,0 +1,247 @@
+// By default, it subscribes to all categories.
+func NewDefaultSubscriber(fmt formatter.Formatter, adp adapter.OutputAdapter) *DefaultSubscriber {
+	return &DefaultSubscriber{
+		id:         utils.GenerateUUID(),
+		categories: []event.EventCategory{event.CategoryAll},
+		eventTypes: make(map[event.EventType]bool),
+		formatter:  fmt,
+		adapter:    adp,
+		logger:     log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName)),
+	}
+}
+
+// NewDefaultSubscriberWithCategories creates a subscriber that filters by categories.
+func NewDefaultSubscriberWithCategories(
+	fmt formatter.Formatter,
+	adp adapter.OutputAdapter,
+	categories ...event.EventCategory,
+) *DefaultSubscriber {
+	if len(categories) == 0 {
+		categories = []event.EventCategory{event.CategoryAll}
+	}
+	return &DefaultSubscriber{
+		id:         utils.GenerateUUID(),
+		categories: categories,
+		eventTypes: make(map[event.EventType]bool),
+		formatter:  fmt,
+		adapter:    adp,
+		logger:     log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName)),
+	}
+}
+
+// NewDefaultSubscriberWithEventTypes creates a subscriber that filters by specific event types.
+func NewDefaultSubscriberWithEventTypes(
+	fmt formatter.Formatter,
+	adp adapter.OutputAdapter,
+	eventTypes ...event.EventType,
+) *DefaultSubscriber {
+	eventTypeMap := make(map[event.EventType]bool)
+	for _, et := range eventTypes {
+		eventTypeMap[et] = true
+	}
+	return &DefaultSubscriber{
+		id:         utils.GenerateUUID(),
+		categories: nil, // Will be determined from event types
+		eventTypes: eventTypeMap,
+		formatter:  fmt,
+		adapter:    adp,
+		logger:     log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName)),
+	}
+}
+
+// GetID returns the unique identifier for this subscriber.
+func (ds *DefaultSubscriber) GetID() string {
+	return ds.id
+}
+
+// GetCategories returns the categories this subscriber is interested in.
+func (ds *DefaultSubscriber) GetCategories() []event.EventCategory {
+	ds.mu.RLock()
+	defer ds.mu.RUnlock()
+
+	if len(ds.categories) > 0 {
+		return ds.categories
+	}
+
+	// If no categories but has event types, derive categories from event types
+	if len(ds.eventTypes) > 0 {
+		categoryMap := make(map[event.EventCategory]bool)
+		for eventType := range ds.eventTypes {
+			category := event.GetCategory(eventType)
+			categoryMap[category] = true
+		}
+
+		categories := make([]event.EventCategory, 0, len(categoryMap))
+		for category := range categoryMap {
+			categories = append(categories, category)
+		}
+		return categories
+	}
+
+	// Default: all categories
+	return []event.EventCategory{event.CategoryAll}
+}
+
+// SetCategories updates the categories this subscriber is interested in.
+func (ds *DefaultSubscriber) SetCategories(categories ...event.EventCategory) {
+	ds.mu.Lock()
+	defer ds.mu.Unlock()
+	ds.categories = categories
+}
+
+// AddEventTypeFilter adds an event type to the filter list.
```

Comment:
The method name does not descriptive enough for me. Feel like this can be improved